server/backend/db_backend.go

```diff
@@ -167,11 +168,36 @@ func (self *MongoBackend) importBlock(ctx context.Context, block *types.Block) (
 	if err != nil {
 		return nil, fmt.Errorf("failed to remove old txs: %v", err)
 	}
+	gasFee := big.NewInt(0)
 	for _, tx := range block.Transactions() {
-		if err := self.importTx(ctx, tx, block); err != nil {
+		if fee, err := self.importTx(ctx, tx, block); err != nil {
 			return nil, fmt.Errorf("failed to import tx: %v", err)
+		} else {
+			gasFee = gasFee.Add(gasFee, fee)
 		}
 	}
+	b.GasFees = gasFee.String()
+	update := bson.M{"gas_fees": b.GasFees}
+	if isDarvaza {
+		parent, err := self.getBlockByHash(b.ParentHash)
+		if err != nil {
+			return nil, err
+		}
+		if parent != nil && parent.TotalFeesBurned != "" {
+			totalFeesBurned, ok := new(big.Int).SetString(parent.TotalFeesBurned, 10)
+			if !ok {
+				lgr.Error("Failed to parse block.TotalFeesBurned as big.Int", zap.String("block", b.ParentHash),
+					zap.String("value", parent.TotalFeesBurned))
+			} else {
+				b.TotalFeesBurned = totalFeesBurned.Add(totalFeesBurned, gasFee).String()
+				update["total_fees_burned"] = b.TotalFeesBurned
```

Comment:
Tracking the cumulative total is easy in the normal case of chronological import where the parent is on hand already.
During a reverse import, when the parent blocks are not available, we just have to trigger some cleanup routine once the chain is 'connected' again.